src/web/extractors.rs

```diff
@@ -838,7 +894,23 @@ impl FromRequest for HawkIdentifier {
 
     /// Use HawkPayload extraction and format as HawkIdentifier.
     fn from_request(req: &HttpRequest, _: &mut Payload) -> Self::Future {
-        Self::extrude(req)
+        let state = match req.app_data::<ServerState>() {
```

Comment:
Just occurred to me now: I suppose we could have a macro for only the future::err case. Whereas non-future returns can opt for the ok_or_else (like I have in get_hawk_id).

(to just throw another alternative to the match statements)

Reply:
I tried that at first, but the problem was wrapping the error types. We'd wind up with several macros (or I suppose adding a block to return the error, but then it felt like "why bother with a macro" so I just reverted back to the matches.